src/linker/Linker/Message.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mono.Linker
+{
+	public class Message
+	{
+		public MessageOrigin Origin { get; }
+		
+		public string Subcategory { get; }
+		
+		public MessageCategory Category { get; }
+		
+		public MessageCode Code { get; }
+		
+		public string Text { get; }
+
+		public Message (
```

Comment:
This constructor is not usable. Why is text optional? Why is origin first when it adds no information by default?

I'd suggest you add only data you know how to pass in. Introducing API without using them is not good start.

Reply:
For the order of the parameters I simply followed the order in which these appear in a MSBuild message (the format is described [here](https://docs.microsoft.com/en-us/archive/blogs/msbuild/msbuild-visual-studio-aware-error-messages-and-message-formats)). `Text` only adds further information for the user about the message he/she's seeing.